### DISCUSSION TITLE
Improve handling of composer `type` field mapping to project/library

### DISCUSSION THREAD

**Discussion by anomiex** [renovatebot/renovate#24807] [Suggest an Idea]
### What would you like Renovate to be able to do?

It's probably not sustainable to keep fixing issues like #13795 by individually adding every custom `type` that someone might use to the hard-coded list. Currently I'm needing "[phpcodesniffer-standard](https://packagist.org/packages/dealerdirect/phpcodesniffer-composer-installer)" and "[jetpack-library](https://packagist.org/packages/automattic/jetpack-composer-plugin)".

It seems I can work around it in our monorepo with packageRules like
```json
        {
            "matchPaths": [...],
            "matchDepTypes": [ "require" ],
            "rangeStrategy": "replace"
        },
        {
            "matchPaths": [...],
            "matchDepTypes": [ "require" ],
            "matchCurrentVersion": "/ \\|\\| /",
            "rangeStrategy": "widen",
        },
```
but having to manually list the paths there would be rather annoying. Even more so that it needs to be done twice to properly handle normal and complex versions in the way "auto" does.

### If you have any ideas on how this should be implemented, please tell us here.

Configuration options like what was suggested in #13795 would make the most sense to me. It's not too clear to me what/where exactly the options should be though.

### Is this a feature you are interested in implementing yourself?

Maybe

**Comment by rarkins** [comment]
We can maybe just make pinning 100% opt in and no more auto mode
